ndc-lambda-sdk/src/inference.ts

```diff
@@ -394,6 +394,18 @@ function isMapType(tsType: ts.Type): boolean {
   return symbol.escapedName === "Map" && symbol.members?.has(ts.escapeLeadingUnderscores("keys")) === true && symbol.members?.has(ts.escapeLeadingUnderscores("values")) === true && symbol.members?.has(ts.escapeLeadingUnderscores("entries")) === true;
 }
 
+// Identifies the 'true | false' type (which is distinct from the 'boolean' type)
+function isBooleanUnionType(tsType: ts.Type): boolean {
+  if (!tsutils.isUnionType(tsType)) return false;
+
+  return tsType.types.length === 2 && unionTypeContainsBothBooleanLiterals(tsType);
+}
+
+function unionTypeContainsBothBooleanLiterals(tsUnionType: ts.UnionType): boolean {
+  return tsUnionType.types.find(tsType => tsutils.isBooleanLiteralType(tsType) && tsType.intrinsicName === "true") !== undefined
+    && tsUnionType.types.find(tsType => tsutils.isBooleanLiteralType(tsType) && tsType.intrinsicName === "false") !== undefined;
+}
+
```

Comment:
Should we also treat `true | undefined` and `false | undefined` in the same way? Just a weird edgecase.

Reply:
That edge case (single boolean literals) is already handled. See in `deriveSchemaTypeIfScalarType` where we use `tsutils.isBooleanLiteralType`. 

The weirdness here is just caused by TypeScript (stupidly?) using the two boolean literals in a union instead of the actual boolean type. They seem equivalent, just two ways of representing the same thing; maybe there's some good reason why they do this.